engine/src/main/java/org/terasology/rendering/dag/nodes/PrePostCompositeNode.java

```diff
@@ -150,6 +148,8 @@ public PrePostCompositeNode(Context context) {
         if (hazeIsEnabled) {
             addDesiredStateChange(setHazeInputTexture);
         }
+
+        addDesiredStateChange(new SwapGBuffers(gBufferPair));
```

Comment:
I guess we'll need to add a warning in the code here, to refer to the SwapGBuffers class and on why this is counterintuitive but correctly placed here.